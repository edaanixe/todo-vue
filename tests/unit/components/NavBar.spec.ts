import { shallowMount } from '@vue/test-utils'
import NavBar from '@/components/NavBar.vue'

describe('NavBar', () => {
    it('should match element against snapshot', () => {
        const wrapper = shallowMount(NavBar)
        expect(wrapper.element).toMatchSnapshot()
    })

    it('should trigger logout event', async () => {
        const wrapper = shallowMount(NavBar)

        await wrapper.find('.btn-logout').trigger('click')

        expect(wrapper.emitted('logout')).toHaveLength(1)
    });
});