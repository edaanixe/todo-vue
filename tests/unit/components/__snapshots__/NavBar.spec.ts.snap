// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match element against snapshot 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
>
  <a
    class="navbar-brand"
    href="#"
  >
    TODO App
  </a>
   
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    class="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
   
  <div
    class="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      class="navbar-nav mr-auto"
    >
      <li
        class="nav-item active"
      >
        <a
          class="nav-link"
          href="#"
        >
          Home 
          <span
            class="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
    </ul>
     
    <form
      class="form-inline my-2 my-lg-0"
    >
      <button
        class="btn my-2 my-sm-0 btn-logout"
        type="submit"
      >
        Logout
      </button>
    </form>
  </div>
</nav>
`;
